addons/hr_holidays/static/src/views/view_dialog/form_view_dialog.js

```diff
@@ -7,12 +7,34 @@ import { registry } from '@web/core/registry';
 import { formView } from '@web/views/form/form_view';
 import { FormController } from '@web/views/form/form_controller';
 
-import { useLeaveCancelWizard } from '../hooks';
+import { useLeaveCancelWizard, useMakeAction } from '../hooks';
 
 export class TimeOffDialogFormController extends FormController {
     setup() {
         super.setup();
         this.leaveCancelWizard = useLeaveCancelWizard();
+        this.makeAction = useMakeAction();
+    }
+
+    async onClick (action) {
+        const record = this.model.root.data;
+        await this.makeAction(action, record.id);
```

Comment:
The hook is not necessary in this case, you can do useService('orm') and do the orm.call in this method